src/python/pants/jvm/run_deploy_jar.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import dataclasses
+import logging
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.core.goals.package import BuiltPackage
+from pants.core.goals.run import RunFieldSet, RunRequest
+from pants.engine.fs import EMPTY_DIGEST, Digest, MergeDigests
+from pants.engine.internals.native_engine import AddPrefix
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.jvm.jdk_rules import JdkSetup, JvmProcess
+from pants.jvm.package.deploy_jar import DeployJarFieldSet
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class __RuntimeJvm:
+    """Allows Coursier to download a JDK into a Digest, rather than an append-only cache for use
+    with `pants run`.
+
+    This is a hideous stop-gap, which will no longer be necessary once `InteractiveProcess` supports
+    append-only caches.
+    """
+
+    digest: Digest
+
+
+@rule(level=LogLevel.DEBUG)
+async def create_deploy_jar_run_request(
+    field_set: DeployJarFieldSet,
+    runtime_jvm: __RuntimeJvm,
```

Comment:
It's not clear how this ends up used... it's being captured, but I don't see anything re-writing the process execution to use it?

Reply:
It gets used here: https://github.com/pantsbuild/pants/pull/14352/files#diff-5ea98966d8a32e9a27419e572c238a03c50be4541df26c93a5ced1c7d08f5351R67